Bottom: 77cb461909c1c1b676bf91f540be9d39ad4032a3
Top:    5c92c87309e0a2b6b296e0c79b67889c98457ab7
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-22 00:23:47 +0100

Refresh of drop-in-pcb-gl-code-various-me

---

diff --git a/src/hid/gtk/gui-pinout-preview.c b/src/hid/gtk/gui-pinout-preview.c
index 53f2a81..80e1288 100644
--- a/src/hid/gtk/gui-pinout-preview.c
+++ b/src/hid/gtk/gui-pinout-preview.c
@@ -155,6 +155,7 @@ pinout_set_data (GhidPinoutPreview * pinout, ElementType * element)
 static gboolean
 ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * ev)
 {
+  extern HID ghid_hid;
   GdkGLContext* pGlContext = gtk_widget_get_gl_context (widget);
   GdkGLDrawable* pGlDrawable = gtk_widget_get_gl_drawable (widget);
   GhidPinoutPreview *pinout = GHID_PINOUT_PREVIEW (widget);
@@ -223,7 +224,7 @@ ghid_pinout_preview_expose (GtkWidget * widget, GdkEventExpose * ev)
 
   /* call the drawing routine */
   ghid_init_triangle_array ();
-  DrawElement (&pinout->element, 0);
+  hid_expose_callback (&ghid_hid, NULL, &pinout->element);
   ghid_flush_triangles ();
 
   if (gdk_gl_drawable_is_double_buffered (pGlDrawable))
